project/src/main/java/com/google/sps/data/MapsLibrary.java

```diff
@@ -1,27 +1,60 @@
 package com.google.sps.data;
 
-/* 
-Basic HTML:
-    <script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY"></script>
-    <body onload="createMap();">
-    <div id="map"></div>
-*/
-
-/*
-Basic CSS:
-    #map {
-        border: thin solid black;
-        height: 500px;
-        width: 500px;
-    }
-*/
-
-/*
-Basic Javascript:
-    function createMap() {
-        const map = new google.maps.Map(
-        document.getElementById('map'),
-        {center: {lat: 37.422, lng: -122.084}, zoom: 16});
+import java.util.stream.Collectors;  
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.sps.data.ProductWithAddress;
+
+import java.util.*; 
+import java.lang.*; 
+
+public class MapsLibrary {
+    private static Double oLat;
+    private static Double oLng;
+    private static Double dLat;
+    private static Double dLng;
+
+    public static Double distance(GeoPt origin, GeoPt destination) {
+        //Convert coordinates to radians instead of degrees
+        oLat = Math.toRadians(Double.parseDouble(new Float(origin.getLatitude()).toString()));
+        oLng = Math.toRadians(Double.parseDouble(new Float(origin.getLongitude()).toString()));
+        dLat = Math.toRadians(Double.parseDouble(new Float(destination.getLatitude()).toString()));
+        dLng = Math.toRadians(Double.parseDouble(new Float(destination.getLongitude()).toString()));
+
+        // Haversine formula
+        Double latDiff = dLat - oLat;
+        Double lngDiff = dLng - oLng;
+        Double ans = Math.pow(Math.sin(latDiff / 2), 2) 
+                  + Math.cos(oLat) * Math.cos(dLat) 
+                  * Math.pow(Math.sin(lngDiff / 2),2);
+        
+        ans = 2 * Math.asin(Math.sqrt(ans));
+
+        //Radius of Earth in miles
+        int radius = 3956;
```

Comment:
you should make this a constant class member.  since it's (hopefully!) not going to change there's no need to create it every time.

`private const int EARTH_RADIUS = 3956;`